tests/functional/i/invalid/invalid_exceptions/invalid_exceptions_raised.py

```diff
@@ -108,3 +108,21 @@ class Error(Exception):
 def bad_case10():
     """raise string"""
     raise "string"  # [raising-bad-type]
+
+
+class AmbiguousValue:
+    """Don't emit when there is ambiguity on the node for the exception."""
+    def __init__(self):
+        self.stored_exception = None
+
+    def fail(self):
+        try:
+            1 / 0
+        except ZeroDivisionError as zde:
+            self.stored_exception = zde
+
+    def raise_stored_exception(self):
+        if self.stored_exception is not None:
+            exc = self.stored_exception
+            self.stored_exception = None
+            raise exc
```

Comment:
I'm wondering what will happen here if we merge the `is None` constraint PR in `astroid`, but that's an issue for another time!